daemon/start.go

```diff
@@ -31,12 +31,22 @@ func (daemon *Daemon) ContainerStart(name string, hostConfig *containertypes.Hos
 		// creating a container, not during start.
 		if hostConfig != nil {
 			logrus.Warn("DEPRECATED: Setting host configuration options when the container starts is deprecated and will be removed in Docker 1.12")
+			oldNetworkMode := container.HostConfig.NetworkMode
 			if err := daemon.setSecurityOptions(container, hostConfig); err != nil {
 				return err
 			}
 			if err := daemon.setHostConfig(container, hostConfig); err != nil {
 				return err
 			}
+			newNetworkMode := container.HostConfig.NetworkMode
+			if string(oldNetworkMode) != string(newNetworkMode) {
```

Comment:
In the case of `docker create`, we call `SetDefaultNetModeIfBlank(hc)` to setup default network mode if it is blank.
This was done as part of the windows support to provide backward compatibility.

With your fix, if the newNetworkMode is empty (which I assume means use the one setup in `docker create`).
Hence, I think we should first check if the `newNetworkMode != ""` before performing the above check.

WDYT ?